h2o-py/tests/testdir_apis/H2OAssembly/pyunit_h2oassembly_to_pojo.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import print_function
+import sys, os
+sys.path.insert(1,"../../")
+import h2o
+from tests import pyunit_utils
+from h2o.utils.typechecks import assert_is_type
+from h2o.assembly import *
+from h2o.transforms.preprocessing import *
+
+def h2oassembly_to_pojo():
+    """
+    Python API test: H2OAssembly.to_pojo(pojo_name=u'', path=u'', get_jar=True)
+
+    Copied from pyunit_assembly_demo.py
+    """
+
+    fr = h2o.import_file(pyunit_utils.locate("smalldata/iris/iris_wheader.csv"),
+                       col_types=["numeric","numeric","numeric","numeric","string"])
+    assembly = H2OAssembly(steps=[("col_select",      H2OColSelect(["sepal_len", "petal_len", "class"])),
+                                ("cos_sep_len",     H2OColOp(op=H2OFrame.cos, col="sepal_len", inplace=True)),
+                                ("str_cnt_species",
+                                 H2OColOp(op=H2OFrame.countmatches, col="class", inplace=False, pattern="s"))])
+
+    result = assembly.fit(fr)  # fit the assembly
+    assert_is_type(result, H2OFrame)
+
+    try:
```

Comment:
avoid try-catch, instead check whether `results_dir` exists explicitly (`os.path.isdir`) and do the correct action based on that